include/circt/Scheduling/Problems.h

```diff
@@ -218,6 +218,17 @@ class Problem {
   }
   void setStartTime(Operation *op, unsigned val) { startTime[op] = val; }
 
+  /// Returns the end time for \p op, as computed by the scheduler.
+  /// This end time is derived from the start time and the operator type's
+  /// latency.
+  Optional<unsigned> getEndTime(Operation *op) {
+    if (auto startTime = getStartTime(op))
+      if (auto opType = getLinkedOperatorType(op))
+        if (auto latency = getLatency(*opType))
+          return startTime.value() + latency.value();
+    return llvm::None;
+  }
+
```

Comment:
I think this is a very useful helper to have, but I'd like to keep this separated from the plain property accessors above. Could you please add a "section" (I mean a banner comment and a `public:`), named "Access to derived properties" or so? Also, should the definition be in the `.cpp` file? I only put one-liners in the header (but am unsure whether the code guidelines agree).